dnf5daemon-server/threads_manager.hpp

```diff
@@ -103,6 +105,63 @@ class ThreadsManager {
         register_thread(std::move(worker));
     }
 
+    template <class S>
+    void handle_method_fd(
+        S & service,
+        void (S::*method)(sdbus::MethodCall &),
+        sdbus::MethodCall & call,
+        std::optional<std::string> thread_locale = std::nullopt) {
+        auto worker = std::thread(
+            [this](
+                S & service,
+                void (S::*method)(sdbus::MethodCall &),
+                sdbus::MethodCall call,
+                std::optional<std::string> thread_locale = std::nullopt) {
+                locale_t new_locale{nullptr};
+                locale_t orig_locale{nullptr};
+                if (thread_locale) {
+                    orig_locale = set_thread_locale(thread_locale.value(), new_locale);
+                }
+
+                const sdbus::MethodReply reply = call.createReply();
+                std::string error_msg{};
+                try {
+                    reply.send();
+                } catch (const std::exception & e) {
+                    error_msg = e.what();
+                } catch (...) {
+                    error_msg = "Unknown exception caught";
+                }
+
+                if (error_msg.empty()) {
+                    try {
+                        (service.*method)(call);
+                    } catch (...) {
+                        // TODO(mblaha): log the error
```

Comment:
Would be possible to add at least `std::cerr << fmt::format(`, but keep TODO?

Reply:
Question is what information could we print here - we do not have an exception (anything could have been thrown), we do not now which method has been called (we only have a pointer to the method).

Reply:
For now we can keep it with TODO. I suggest to log general unique message that will allow us to identify the correct catch in the code. If I am not mistaken it is possible to get something like a traceback with C++20.